src/semafour.cc

```diff
@@ -76,11 +79,11 @@ int Semafour::Post() {
 }
 
 
-int Semafour::Wait() {
+int Semafour::Wait(bool nonBlocking) {
```

Comment:
Please don't combine `sem_wait()` and `sem_trywait()` in the same function.

Reply:
Should I attempt to avoid repetition at all or should I just have a slightly redundant function?

Reply:
I'd keep them as completely separate functions. It's only a few lines, and it makes debugging simpler.

Reply:
Got it!